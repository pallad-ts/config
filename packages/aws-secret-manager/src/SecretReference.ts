export interface SecretReference {
	name: string;
	property?: string;
}

export namespace SecretReference {
	export function computeDescription({ name, property }: SecretReference) {
		return `AWS Secret Manager secret name: ${name}${property ? `, property: ${property}` : ""}`;
	}
}
