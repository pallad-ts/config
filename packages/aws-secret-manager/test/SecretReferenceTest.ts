import { SecretReference } from "@src/SecretReference";

describe("SecretReference", () => {
	it.each<[SecretReference]>([
		[
			{
				name: "s1",
			},
		],
		[
			{
				name: "s2",
				property: "p2",
			},
		],
	])("description %#", ref => {
		expect(SecretReference.computeDescription(ref)).toMatchSnapshot();
	});
});
