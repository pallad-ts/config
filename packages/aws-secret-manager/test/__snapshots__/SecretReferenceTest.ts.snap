// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecretReference description 0 1`] = `"AWS Secret Manager secret name: s1"`;

exports[`SecretReference description 1 1`] = `"AWS Secret Manager secret name: s2, property: p2"`;
